### ISSUE TITLE
Feature request: export current language

### ISSUE THREAD

**Issue by alexvoss** [ultrabug/mkdocs-static-i18n#75]
Hi,

I am writing a macro using mkdocs-macros-plugin that is to include content depending on the language chosen but I could not find a way to identify the language outside your plugin. Is there a chance you could export it? Would be a two-liner to add an instance variable or just a couple more to add a getter.

Thanks for a all the great work on the plugin, it works a treat!

Cheers,

Alex

**Comment by alexvoss** [comment]
Sorry, can we add the default language as well? When I request env.conf['plugins']['i18n'] in my macro, I get the I18n plugin object and not the content of mkdocs.yml

**Comment by alexvoss** [comment]
Hi @ultrabug, 

if you let me know that I am not holding the wrong end of the stick, I could submit a pull request.

Cheers,

Alex

**Comment by ultrabug** [comment]
Hello @alexvoss ; yes I guess this is absolutely doable

I guess you'd like to get it on the page context right ? Injecting stuff in env looks fine

If you're up for it, even better! thanks

**Comment by ultrabug** [comment]
I added `i18n_config` and `i18n_page_locale`, hope this is useful enough mate

**Comment by alexvoss** [comment]
Hi, I can see you added these but not sure there they can be picked up on my side? Ideally, I would be able to see the variables in  `env.variables` in my mkdocs-macro. I tried to figure out just what the 'context' object is but no joy. Where does `on_page_context` even get called from and where does the `context` come from and go to? Sorry, the documentation is somewhat thin on this and I cannot make sense of the code as I cannot find where the call happens.

**Comment by ultrabug** [comment]
@alexvoss I have released 0.31 with some added variables, fixed logic and a better documentation on how to use it : https://github.com/ultrabug/mkdocs-static-i18n#using-i18n-context-variables-in-your-pages

**Comment by alexvoss** [comment]
Thanks for this. I can see that it is possible to use the variables in a template. Unfortunately, my use case requires their use from [mkdocs-macros](https://mkdocs-macros-plugin.readthedocs.io/en/latest/) and I cannot figure out how to access the same variables from my macros in the page content. They are undefined while the same variables are defined when the template processing happens. 

If this is to do with the fundamental logic of how the page processing happens, would it be possible to expose the same variables where I can gain access to them, e.g., as a variable or getter on the plugin object itself? I can get access to that easily enough using `env.conf["plugins"]["i18n"]`. This is the way I implemented this before I opened the feature request: 

```
self.page_locale = page.file.dest_language or self.default_language
self.page_file_locale = page.file.locale_suffix
```

**Comment by ultrabug** [comment]
Well @alexvoss you can access them through the `page` variable which is [included by mkdocs macros](https://mkdocs-macros-plugin.readthedocs.io/en/latest/pages/#batteries-included-defaut-objects)

`page.file.locale_suffix` for instance is the same as `page_file_locale` in the jinja2 context. Everything part of the I18nFile object will be accessible through `page.file.xxx`, [the attributes of file can be seen here](https://github.com/ultrabug/mkdocs-static-i18n/blob/main/mkdocs_static_i18n/struct.py#L97).

Does that help?

**Comment by alexvoss** [comment]
Yes, that does help! Thanks a lot. Here is what I ended up with:

```
def get_lang(env):
    lang = env.page.file.dest_language
    i18n = env.conf['plugins']['i18n']
    return lang if lang and lang != "" else i18n.default_language
```
Does this look sensible to you?

I do wonder why I did not find `page.file`. It does not seem to be listed in the [description of the page object](https://www.mkdocs.org/dev-guide/themes/#page) - unless I am missing something. I'll double check and submit a bug report with MkDocs if it looks like I am right.
